bliss/surveys/dc2.py

```diff
@@ -262,33 +382,43 @@ def from_file(cls, cat_path, wcs, height, width, band):
             "star_log_fluxes": star_log_fluxes.reshape(1, nobj, 6),
         }
 
-        return cls(height, width, d), torch.stack(psf_params)
+        if return_match_id:
+            return cls(height, width, d), torch.stack(psf_params), match_id
+        else:
+            return cls(height, width, d), torch.stack(psf_params)
 
 
 def read_frame_for_band(image_files, bg_files, n, n_bands, image_lim):
     image_list = []
     bg_list = []
     wcs = None
+    wcs_header_str = None
     for b in range(n_bands):
         image_frame = fits.open(image_files[b][n])
         bg_frame = fits.open(bg_files[b][n])
         image_data = image_frame[1].data  # pylint: disable=maybe-no-member
         bg_data = bg_frame[0].data  # pylint: disable=maybe-no-member
 
         if wcs is None:
-            wcs = WCS(image_frame[1].header)  # pylint: disable=maybe-no-member
+            wcs_header = image_frame[1].header
+            wcs = WCS(wcs_header)  # pylint: disable=maybe-no-member
+            wcs_header_str = wcs_header.tostring()
 
         image_frame.close()
         bg_frame.close()
 
         image = torch.nan_to_num(torch.from_numpy(image_data)[: image_lim[0], : image_lim[1]])
         bg = torch.from_numpy(bg_data.astype(np.float32)).expand(image_lim[0], image_lim[1])
 
+        # to make sure the background are positive
+        bg = bg.clone()
+        bg[bg <= 1e-6] = 1e-5
+
         image += bg
         image_list.append(image)
         bg_list.append(bg)
 
-    return image_list, bg_list, wcs
+    return image_list, bg_list, wcs, wcs_header_str
```

Comment:
Can we avoid returning both `wcs` and `wcs_header_str`?

Reply:
I return `wcs_header_str` because i want to ensure we can rebuild the same wcs from header string. In wcs's document, we can find the warning w.r.t `to_header` and `to_header_string` that 

> This function does not write out FITS WCS [distortion paper](https://www.atnf.csiro.au/people/mcalabre/WCS/dcs_20040422.pdf) information, since that requires multiple FITS header data units. To get a full representation of everything in this object, use [to_fits](https://docs.astropy.org/en/stable/api/astropy.wcs.WCS.html#astropy.wcs.WCS.to_fits).

Therefore, if we only return wcs and use it to generate header string, we may lose some information. Maybe we can just return `wcs_header_str`.

Reply:
I wonder if the distortion paper information is important...perhaps not if it doesn't get stored to the WCS object. In any case, it could be fine to only return `wcs_header_str`

Reply:
We now only return `wcs_header_str` in commit a521a09.